src/common/snippets/src/lowered/loop_info.cpp

```diff
@@ -187,34 +178,34 @@ bool UnifiedLoopInfo::LoopPortDesc::is_dynamic() const {
 
 UnifiedLoopInfo::UnifiedLoopInfo(size_t work_amount, size_t increment,
                                  const std::vector<LoopPort>& entries, const std::vector<LoopPort>& exits,
-                                 const SpecificIterationHandlers& handlers, bool is_wa_const)
-    : LoopInfo(work_amount, increment, entries, exits, is_wa_const), m_handlers(handlers),
+                                 const SpecificIterationHandlers& handlers)
+    : LoopInfo(work_amount, increment, entries, exits), m_handlers(handlers),
       m_input_port_descs(std::vector<LoopPortDesc>(entries.size())), m_output_port_descs(std::vector<LoopPortDesc>(exits.size())) {
     sort_ports();
 }
 
 UnifiedLoopInfo::UnifiedLoopInfo(size_t work_amount, size_t increment,
                                  const std::vector<ExpressionPort>& entries, const std::vector<ExpressionPort>& exits,
-                                 const SpecificIterationHandlers& handlers, bool is_wa_const)
-    : LoopInfo(work_amount, increment, entries, exits, is_wa_const), m_handlers(handlers),
+                                 const SpecificIterationHandlers& handlers)
+    : LoopInfo(work_amount, increment, entries, exits), m_handlers(handlers),
       m_input_port_descs(std::vector<LoopPortDesc>(entries.size())), m_output_port_descs(std::vector<LoopPortDesc>(exits.size())) {
     sort_ports();
 }
 
 UnifiedLoopInfo::UnifiedLoopInfo(size_t work_amount, size_t increment,
                                  const std::vector<LoopPort>& entries, const std::vector<LoopPort>& exits,
                                  const std::vector<LoopPortDesc>& in_shifts, const std::vector<LoopPortDesc>& out_shifts,
-                                 const SpecificIterationHandlers& handlers, bool is_wa_const)
-    : LoopInfo(work_amount, increment, entries, exits, is_wa_const), m_handlers(handlers), m_input_port_descs(in_shifts), m_output_port_descs(out_shifts) {
+                                 const SpecificIterationHandlers& handlers)
+    : LoopInfo(work_amount, increment, entries, exits), m_handlers(handlers), m_input_port_descs(in_shifts), m_output_port_descs(out_shifts) {
     sort_ports();
 }
 
-std::shared_ptr<LoopInfo> UnifiedLoopInfo::clone_with_new_expr(const ExpressionMap& expr_map) const {
+std::shared_ptr<LoopInfo> UnifiedLoopInfo::clone_with_new_expr(const ExpressionMap& expr_map, LoopInfoMap& loop_map) const {
```

Comment:
Can `this` already be in the `LoopInfoMap`?
From the algorithm correctness point of view, the same question holds for expanded and splitter loop infos.

Reply:
Done, thanks